dim/src/scanners/base.rs

```diff
@@ -356,9 +360,66 @@ impl MetadataMatcher {
 
     #[handler]
     pub async fn match_tv(&mut self, media: MediaFile) -> Result<(), ScannerError> {
-        let mut media = media;
+        let mut media_file = media;
+
+        {
+            let mut tx = self.conn.read().begin().await.unwrap();
+
+            match Media::get_of_mediafile(&mut tx, media_file.id).await {
+                Ok(media) => {
+                    assert_eq!(media.media_type, MediaType::Episode);
+
+                    let episode = Episode::get_by_id(&mut tx, media.id).await.unwrap();
```

Comment:
I think we might want to write a `Episode::get_of_mediafile` function because `Episode::get_by_id` will in itself call another db function.

Basically we end up with 3 db calls here when we could just have one

Reply:
sure, can I leave this to you for now? I don't want to get elbow deep in SQL at the moment.